gowrapper/main.go

```diff
@@ -110,13 +111,45 @@ func main() {
 	})
 }
 
-func newJSDevice(onWrite func([]byte)) *js.Object {
+type readWriteCloser struct {
+	read  func(p []byte) (n int, err error)
+	write func(p []byte) (n int, err error)
+}
+
+func (r readWriteCloser) Read(p []byte) (n int, err error) {
+	return r.read(p)
+}
+
+func (r readWriteCloser) Write(p []byte) (n int, err error) {
+	return r.write(p)
+}
+
+func (r readWriteCloser) Close() error { return nil }
+
+// useBitBoxBridge:
+// - if true, assume the connection is to the BitBoxBridge, which uses a slightly different communication protocol (no u2fhid frames).
+// - if false, assume the connection is via WebHID, so the regular u2fhid framing protocol is used.
+func newJSDevice(onWrite func([]byte), useBitBoxBridge bool) *js.Object {
```

Comment:
couldn't find this function in https://github.com/digitalbitbox/bitbox02-api-js/blob/master/docs/methods.md.

Reply:
It is not part of the public api - bitbox02.js contains the public API and it uses newJSDevice internally.

Reply:
`js.Global.Set("bitbox02", ..., "New": newJSDevice, ...)` - is this not public? https://github.com/digitalbitbox/bitbox02-api-js/blob/cbcb920/gowrapper/main.go#L79

Reply:
No, only what's in `index.js` is public, which is mostly `BitBox02API`.

Reply:
then I would split this into 2 functions: `newBridgedDevice` and `newWebHIDDevice`.

Reply:
done, ptal